crates/core/src/util/lending_pool.rs

```diff
@@ -80,9 +80,10 @@ impl<T> LendingPool<T> {
         queue_len_max.set(max_queue_len);
 
         async move {
-            let permit = acq.await.map_err(|_| PoolClosed)?;
+            let permit_result = acq.await.map_err(|_| PoolClosed);
             queue_len.dec();
```

Comment:
I might wrap this in a scope guard to future proof. At the very least I would add a comment about decrementing in the case of a semaphore closure.